### PR TITLE
update: set to use build and platform for reproduce comparision job and remove cleanWs pre-build.

### PR DESCRIPTION
**Pull request by zdtsw**
3 parts in this PR:
- Fix: #395 
- add better info. when build failed with different error
- some test on mac part to fix permission on script and remove pre-cleanup workspace so we dont have script removed....

### CODE REVIEW

File: tools/reproduce_comparison/Jenkinsfile [hunk 1]
```diff
@@ -14,27 +14,34 @@ pipeline {
     stages {
         stage('Lets Compare') {
             agent {
-                label "${params.platform}"
-            }
+                label "${params.platform}&&build"
```
Comment (karianna) [inline tools/reproduce_comparison/Jenkinsfile:17]:
```suggestion
                label "${params.platform}&&build" // build label enables comparisons to be done
```

File: tools/reproduce_comparison/Jenkinsfile [hunk 2]
```diff
@@ -14,27 +14,34 @@ pipeline {
...
-                    cleanWs()
                     try {
                         echo "Fetching artifact1 from ${params.URL1}"
-                        sh "mkdir url1Dir1 && wget -q ${params.URL1} -O - | tar -xz -C url1Dir1"
+                        def ret1 = sh returnStatus: true, script: "mkdir url1Dir1 && wget -q ${params.URL1} -O - | tar -xz -C url1Dir1"
                         echo "Fetching artifact2 from ${params.URL2}"
-                        sh "mkdir url1Dir2 && wget -q ${params.URL2} -O - | tar -xz -C url1Dir2"
+                        def ret2 = sh returnStatus: true, script: "mkdir url1Dir2 && wget -q ${params.URL2} -O - | tar -xz -C url1Dir2"
+                        if (ret1 || ret2) { 
+                            currentBuild.result = "UNSTABLE"
```
Comment (karianna) [inline tools/reproduce_comparison/Jenkinsfile:27]: recommend adding more logging so we know which of the two failed.

File: tools/reproduce_comparison/Jenkinsfile [hunk 3]
```diff
@@ -14,27 +14,34 @@ pipeline {
...
-            }
+                label "${params.platform}&&build"
+            } 
             steps {
                 script {
-                    cleanWs()
                     try {
                         echo "Fetching artifact1 from ${params.URL1}"
-                        sh "mkdir url1Dir1 && wget -q ${params.URL1} -O - | tar -xz -C url1Dir1"
+                        def ret1 = sh returnStatus: true, script: "mkdir url1Dir1 && wget -q ${params.URL1} -O - | tar -xz -C url1Dir1"
```
Comment (karianna) [inline tools/reproduce_comparison/Jenkinsfile:23]: How would returnStatus ever get set to false?

File: tools/reproduce_comparison/Jenkinsfile [hunk 4]
```diff
@@ -14,27 +14,34 @@ pipeline {
...
                         // call extra platform specific function
                         if ( params.platform != 'linux' ) { 
                             "prep${platform}" ("url1Dir1", "url1Dir2") 
                         }
-                        retVal = sh returnStatus: true, script: "diff -q -r url1Dir1 url1Dir2"
+                        def retVal = sh returnStatus: true, script: "diff -q -r url1Dir1 url1Dir2"
                         if (retVal != 0 ){
-                            echo "Error: two builds are not the same!"
                             currentBuild.result = 'FAILURE'
+                            error "Error: two builds are not the same!"
```
Comment (karianna) [inline tools/reproduce_comparison/Jenkinsfile:38]: Might be worth adding the diff output

**Review by karianna — CHANGES_REQUESTED** [review]

File: tools/reproduce_comparison/Jenkinsfile (hunk 3 above)
Comment (zdtsw) [reply tools/reproduce_comparison/Jenkinsfile:23]: you are right! i need to fix this

File: tools/reproduce_comparison/Jenkinsfile (hunk 2 above)
Comment (zdtsw) [reply tools/reproduce_comparison/Jenkinsfile:27]: so the step "sh" prints out error already, e.g 
```
+ mkdir url1Dir2
+ wget -q https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk18-pipeline/256/artifact/target/linux/x64/temurin/OpenJDK18U-jdk_x64_linux_hotspot_2022-09-03-23-30.tar.gz -O -
+ tar -xz -C url1Dir2

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```

File: tools/reproduce_comparison/Jenkinsfile (hunk 4 above)
Comment (zdtsw) [reply tools/reproduce_comparison/Jenkinsfile:38]: from test https://ci.adoptopenjdk.net/view/work-in-progress/job/wen_reproducible_comparision/9/console
it lists all the files before prints `Error: two builds are not the same!`

**Review by karianna — APPROVED** [review]

**Review by steelhead31 — APPROVED** [review]
LGTM

**Pull request merged**